README.md

```diff
@@ -16,7 +16,384 @@ SPDX-License-Identifier: MIT
+
+The module automatically handles task persistence and versioning. When you update a task's schedule or details, it creates a new version without affecting previously completed events. This ensures your historical data remains intact.
+
+You create tasks using `createOrUpdateTask()`, and the module takes care of generating the appropriate events based on your schedule.
+
+### Setup
+
+You need to add the Spezi Scheduler Swift package to
+[your app in Xcode](https://developer.apple.com/documentation/xcode/adding-package-dependencies-to-your-app) or
+[Swift package](https://developer.apple.com/documentation/xcode/creating-a-standalone-swift-package-with-xcode#Add-a-dependency-on-another-Swift-package).
+
+> [!IMPORTANT]  
+> If your application is not yet configured to use Spezi, follow the [Spezi setup article](https://swiftpackageindex.com/stanfordspezi/spezi/documentation/spezi/initial-setup) to set up the core Spezi infrastructure.
+
+Below is an example on how to create your own [`Module`](https://swiftpackageindex.com/stanfordspezi/spezi/documentation/spezi/module) to manage your tasks and ensure they are always up to date.
+
+```swift
+import Spezi
+import SpeziScheduler
+
+class MySchedulerModule: Module {
+    @Dependency(Scheduler.self)
+    private var scheduler
+
+    init() {}
+
+    func configure() {
+        do {
+            try scheduler.createOrUpdateTask(
+                id: "my-daily-task",
+                title: "Daily Questionnaire",
+                instructions: "Please fill out the Questionnaire every day.",
+                category: .questionnaire,
+                schedule: .daily(hour: 9, minute: 0, startingAt: .today)
+            )
+        } catch {
+            // handle error (e.g., visualize in your UI)
+        }
+    }
+}
+```
+
+### Creating and Managing Tasks
+
+Then, configure the [`Scheduler`](https://swiftpackageindex.com/stanfordspezi/spezischeduler/documentation/spezischeduler/scheduler) module and your custom module in your `SpeziAppDelegate`:
+```swift
+class ExampleAppDelegate: SpeziAppDelegate {
+    override var configuration: Configuration {
+        Configuration(standard: ExampleStandard()) {
+            MySchedulerModule()
+            Scheduler()
+        }
+    }
+}
+```
+
+### Task Scheduling Options
+
+The Scheduler supports various scheduling patterns using the [`Schedule`](https://swiftpackageindex.com/stanfordspezi/spezischeduler/documentation/spezischeduler/schedule) type:
+
+```swift
+// One-time task
+let onceSchedule = Schedule.once(at: Date(), duration: .tillEndOfDay)
+
+// Daily tasks
+let dailySchedule = Schedule.daily(hour: 8, minute: 30, startingAt: .today)
+
+// Weekly tasks
+let weeklySchedule = Schedule.weekly(
+    weekday: .monday, 
+    hour: 10, 
+    minute: 0, 
+    startingAt: .today
+)
+
+// Monthly tasks
+let monthlySchedule = Schedule.monthly(
+    day: 1, 
+    hour: 9, 
+    minute: 0, 
+    startingAt: .today
+)
```

Comment:
i'd probably write these as eg `let dailySchedule: Schedule = .daily(...)`; that's also how you'd use the API when calling `createOrUpdateTask` (via the dot-syntax thing)